solidity/random-beacon/contracts/libraries/Groups.sol

```diff
@@ -148,29 +127,186 @@ library Groups {
+        return self.groupsData[self.groupsRegistry[groupId]].terminated;
+    }
+
+    /// @notice Gets the cutoff time until which the given group is considered
+    ///         to be active assuming it hasn't been terminated before.
+    function groupLifetimeOf(Data storage self, bytes32 groupPubKeyHash)
+        internal
+        view
+        returns (uint256)
+    {
+        return
+            self.groupsData[groupPubKeyHash].activationBlockNumber +
+            self.groupLifetime;
+    }
+
+    /// @notice Gets the cutoff time in blocks after which the given group is
+    ///         considered as stale. Stale group is an expired group which is no
+    ///         longer performing any operations.
+    function groupStaleTime(Data storage self, bytes32 groupPubKeyHash)
+        internal
+        view
+        returns (uint256)
+    {
+        return groupLifetimeOf(self, groupPubKeyHash) + self.relayEntryTimeout;
+    }
+
+    /// @notice Checks if a group with the given index is a stale group.
+    ///         Stale group is an expired group which is no longer performing any
+    ///         operations. It is important to understand that an expired group
+    ///         may still perform some operations for which it was selected when
+    ///         it was still active. We consider a group to be stale when it's
+    ///         expired and when its expiration time and potentially executed
+    ///         operation timeout are both in the past.
+    function isStaleGroup(Data storage self, bytes memory groupPubKey)
+        internal
+        view
+        returns (bool)
+    {
+        // TODO: Can a group be considered as "stale" if "expiredGroupOffset" was
+        //       not moved forward which means that group selection wasn't triggered?
+        //       In other words can it be stale when it's not officially expired?
+        return groupStaleTime(self, keccak256(groupPubKey)) < block.number;
+    }
+
+    /// @notice Checks if a group with the given index is a stale group.
+    ///         Stale group is an expired group which is no longer performing any
+    ///         operations. It is important to understand that an expired group
+    ///         may still perform some operations for which it was selected when
+    ///         it was still active. We consider a group to be stale when it's
+    ///         expired and when its expiration time and potentially executed
+    ///         operation timeout are both in the past.
+    function isStaleGroup(Data storage self, uint64 groupId)
+        internal
+        view
+        returns (bool)
+    {
+        // TODO: Can a group be considered as "stale" if "expiredGroupOffset" was
+        //       not moved forward (group selection wasn't triggered)?
+        //       In other words can it be stale when it's not officially expired?
+        return
+            groupStaleTime(self, self.groupsRegistry[groupId]) < block.number;
     }
```

Comment:
Are those methods used anywhere?

Reply:
Not for now, but I'm making ground for future use: rewards and reporting unauthorized signing. Also, I [confirmed](https://discordapp.com/channels/893441201628405760/893441585881161738/905397883782701106) on Discord that we'll need all 4 statuses: active, terminated, expired, and stale.